src/lib/store.js

```diff
@@ -499,124 +411,41 @@ export default class Store {
 		this.storeUpdate();
 	};
 
-	updateVendorList = vendorList => {
-		const {
-			pubVendorsList = {},
-			allowedVendorIds
-		} = this;
-
+	updateVendorList = (vendorList) => {
 		const {
 			created,
-			maxVendorId = 0
-		} = this.vendorConsentData;
-
-		if (vendorList) {
-			// Filter vendors in vendorList by allowedVendorIds
-			if (vendorList.vendors && allowedVendorIds.size) {
-				vendorList.vendors = vendorList.vendors.filter(({id}) => allowedVendorIds.has(id));
-			}
-
-			// If a pubVendorList is applied make the vendor list version = 0
-			const { publisherVendorsVersion } = pubVendorsList;
-			vendorList.vendorListVersion = publisherVendorsVersion ? 0 : vendorList.vendorListVersion;
-		}
+		} = this.persistedConsentData || {};
 
 		const {
-			vendors = [],
-			purposes = [],
+			vendors = {},
 		} = vendorList || {};
 
-		//collection of global and local vendor ids [global id, local id]
-		this.globalVendorIdsPresentOnList = new Map(vendors.filter(vendor => vendor.external_id).map(
-			vendor => [vendor.external_id, vendor.id]
-		));
+		const persistedMaxVendorId = this.persistedConsentData.vendorConsents && this.persistedConsentData.vendorConsents.maxId || 0;
+
+		this.tcModel.gvl = new GVL(vendorList);
+		this.cmpApi.tcModel = this.tcModel;
+		this.vendorList = vendorList;
 
-		// If vendor consent data has never been persisted set default selected status
+		// If vendor and publisher consent data has never been persisted set default selected status
 		if (!created) {
-			this.vendorConsentData.selectedPurposeIds = new Set(purposes.map(p => p.id));
-			this.vendorConsentData.selectedVendorIds = new Set(vendors.map(v => v.id));
+			this.selectAllPurposes(true);
+			this.selectAllPurposesLegitimateInterests(true);
+			this.selectAllVendors(true);
+			this.selectAllVendorLegitimateInterests(true);
+			this.selectAllSpecialFeatureOptins(true);
+
+			this.selectAllPublisherPurposes(true);
+			this.selectAllPublisherLegitimateInterests(true);
```

Comment:
We notify store update listeners 7 times here. Is this OK?

Reply:
Good point, fixed.